src/components/Widget/layout.tsx

```diff
@@ -52,32 +54,88 @@ function WidgetLayout({
   launcherCloseLabel,
   sendButtonAlt,
   showTimeStamp,
-  imagePreview
+  imagePreview,
+  zoomStep,
 }: Props) {
+  const dispatch = useDispatch();
   const { dissableInput, showChat } = useSelector((state: GlobalState) => ({
     showChat: state.behavior.showChat,
     dissableInput: state.behavior.disabledInput
   }));
 
+  const messageRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if(showChat) {
+      messageRef.current = document.getElementById('messages') as HTMLDivElement;
+    }
+    return () => {
+      messageRef.current = null;
+    }
+  }, [showChat])
+  
+  const eventHandle = evt => {
+    if(evt.target && evt.target.className === 'rcw-message-img') {
+      const { src, alt, naturalWidth, naturalHeight } = (evt.target as HTMLImageElement);
+      const obj = {
+        src: src,
+        alt: alt,
+        width: naturalWidth,
+        height: naturalHeight,
+      };
+      dispatch(openFullscreenPreview(obj))
+    }
+  }
+
+  /**
+   * Previewer needs to prevent body scroll behavior when fullScreenMode is true
+   */
+  useEffect(() => {
+    const target = messageRef?.current;
+    if(imagePreview && showChat) {
+      target?.addEventListener('click', eventHandle, false);
+    }
+
+    return () => {
+      target?.removeEventListener('click', eventHandle);
+    }
+  }, [imagePreview, showChat]);
+
+  useEffect(() => {
+    if(fullScreenMode) {
+      document.body.setAttribute('style', "overflow: hidden")
+    } else {
+      document.body.setAttribute('style', "overflow: auto")
+    }
+  }, [fullScreenMode])
+
   return (
-    <div className={cn('rcw-widget-container', { 'rcw-full-screen': fullScreenMode })}>
-      <Conversation
-        title={title}
-        subtitle={subtitle}
-        sendMessage={onSendMessage}
-        senderPlaceHolder={senderPlaceHolder}
-        profileAvatar={profileAvatar}
-        toggleChat={onToggleConversation}
-        showCloseButton={showCloseButton}
-        disabledInput={dissableInput}
-        autofocus={autofocus}
-        titleAvatar={titleAvatar}
-        className={showChat ? 'active' : 'hidden'}
-        onQuickButtonClicked={onQuickButtonClicked}
-        onTextInputChange={onTextInputChange}
-        sendButtonAlt={sendButtonAlt}
-        showTimeStamp={showTimeStamp}
-      />
+    <div
+      className={cn('rcw-widget-container', {
+        'rcw-full-screen': fullScreenMode,
+        'rcw-previewer': imagePreview
+        })
+      }
+    >
+      {showChat &&
```

Comment:
bug fixed here.
but animation is not working.
I am looking for the solution

Reply:
Was the issue about the render of the component? The animation can be made via CSS and control the render of the component via state variables 🤔

Reply:
yep, I do this already. I try to use JS to control DOM, but it's not best solution for performance. So we can leave it to another version though ?

Reply:
Of course, animations are pretty rough so those are something to work on 😄